### ISSUE TITLE
[BUG]: Fix dlp-cli so that it has better regex for dlp env

### ISSUE THREAD

**Issue by dwu359** [DSGT-DLP/Deep-Learning-Playground#988]
### Operating system

Noah's machine

### Bug description

The dlp-cli will detect a false positive for a mamba env if the mamba env contains the string "dlp" in it, even if the env is named differently.

### Steps to reproduce

Create a mamba env with a name containing "dlp", install backend via dlp-cli

**Comment by karkir0003** [comment]
@dwu359 this bug isn't just on Noah's machine. It seems to be across diff OS

**Comment by dwu359** [comment]
Yep I know, that's just the OS where we discovered the bug, and it definitely isn't an OS dependent bug
